# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../pptpd/pptpd-1.3.4-pptpctrl-remotenumber.patch
# Copyright (C) 2011 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From f8f1373c079c9108b283255d39c0ccc083174b4f Mon Sep 17 00:00:00 2001
From: Christian Wiese <christian.wiese@securepoint.de>
Date: Wed, 24 Aug 2011 13:37:13 +0200
Subject: [PATCH] pptpctrl: changed to call pppd with remotenumber command line argument

Note:

This will workaround a problem when using the RADIUS plugin in pppd, it send
Caller-Station-ID attribuse to radius server, but pppd needs to know remote
user IP.

Origin: http://patch-tracker.debian.org/patch/misc/dl/pptpd/1.3.4-5/pptpctrl.c

http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=610375
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=625459
---
 pptpctrl.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git pptpd-1.3.4.orig/pptpctrl.c 1.3.4/pptpctrl.c
index b4e52dc..480827b 100644
--- pptpd-1.3.4.orig/pptpctrl.c
+++ 1.3.4/pptpctrl.c
@@ -657,7 +657,7 @@ static int startCall(char **pppaddrs, struct in_addr *inetaddrs)
  */
 static void launch_pppd(char **pppaddrs, struct in_addr *inetaddrs)
 {
-	char *pppd_argv[14];
+	char *pppd_argv[16];
 	int an = 0;
 	sigset_t sigs;
 
@@ -777,6 +777,9 @@ static void launch_pppd(char **pppaddrs, struct in_addr *inetaddrs)
                  pppd_argv[an++] = "pptpd-original-ip";
                  pppd_argv[an++] = inet_ntoa(inetaddrs[1]);
         }
+        
+        pppd_argv[an++] = "remotenumber";
+        pppd_argv[an++] = inet_ntoa(inetaddrs[1]);
 
 	/* argv arrays must always be NULL terminated */
 	pppd_argv[an++] = NULL;
-- 
1.6.6.2

